src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -1072,6 +1074,12 @@ public ResultSet getColumns(String c, String s, String tblNamePattern, String co
                             }
                         }
 
+                        // remove (optional) length/ dimension of the column before returning result set
+                        Matcher regexMatcher = TYPE_WITH_SIZE.matcher(colType);
+                        if (regexMatcher.find()) {
+                            colType = regexMatcher.replaceFirst("$1").trim();
```

Comment:
I'm a bit dubious as to why this syntax is used, why not just use `regexMatcher.group(1).trim()` ?